### PR TITLE
chore(environments): Calculate cohorts for each environment

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

Appendix A of the [environments implementation plan](https://github.com/PostHog/posthog/issues/13418#issuecomment-2180883524).

## Changes

`recalculate_cohortpeople()` and `clear_stale_cohortpeople()` now insert and clear cohort people for every team the cohort is available in, instead of just the one the cohort was created in.

## How did you test this code?

Added a test case to `test_cohort.py`. Definitely let me know how to improve test coverage!

### CODE REVIEW

File: posthog/api/cohort.py [hunk 1]
```diff
@@ -240,22 +238,31 @@ def update(self, cohort: Cohort, validated_data: dict, *args: Any, **kwargs: Any
 
         is_deletion_change = deleted_state is not None and cohort.deleted != deleted_state
         if is_deletion_change:
+            relevant_team_ids = (
+                Team.objects.order_by("id").filter(project_id=cohort.team.project_id).values_list("id", flat=True)
```
Comment (anirudhpillai) [inline posthog/api/cohort.py:242]: curious why we need `order_by("id")` here

**Review by anirudhpillai — COMMENTED** [review]
Looks ok to me but will let Sandy comment on how this fits in with his rewrite + maybe @dmarticus as feature flags uses behavioural cohorts

File: posthog/api/cohort.py (hunk 1 above)
Comment (Twixes) [reply posthog/api/cohort.py:242]: Good question, I don't remember 😄 It's definitely superfluous here, removed

**Review by aspicer — APPROVED** [review]
lgtm!

**Pull request merged**